### ISSUE TITLE
`inconsistent_struct_constructor` seems completely broken

### ISSUE THREAD

**Issue by cyqsimon** [rust-lang/rust-clippy#16020]
### Summary

Using the example provided in [lint docs](https://rust-lang.github.io/rust-clippy/stable/index.html#inconsistent_struct_constructor), I cannot get `inconsistent_struct_constructor` to ever emit anything, locally or in [Rust Playground](https://play.rust-lang.org/?version=stable&mode=release&edition=2024&gist=e5431b76c64aa67cc87df10747558c02).

I've tested 1.65.0, 1.77.2, and 1.91.0 (pseudo-randomly chosen; no particular reason). Not really sure what's going on here.

### Lint Name

inconsistent_struct_constructor

### Reproducer

```rs
#![allow(unused)]
#![warn(clippy::inconsistent_struct_constructor)]

fn main() {
    struct Foo {
        x: i32,
        y: i32,
    }
    let x = 1;
    let y = 2;
    
    Foo { y, x };
}
```

Observe that clippy does not emit a warning.

### Version

```text
rustc 1.91.0 (f8297e351 2025-10-28)
binary: rustc
commit-hash: f8297e351a40c1439a467bbbb6879088047f50b3
commit-date: 2025-10-28
host: x86_64-unknown-linux-gnu
release: 1.91.0
LLVM version: 21.1.2
```

**Comment by Jarcho** [comment]
Your playground link is emitting `inconsistent_struct_constructor`.

**Comment by cyqsimon** [comment]
This is the most bizarre situation I've found myself in in a good while. I swear it wasn't working when I posted 🤨

Please allow me some time to see what the hell just happened

**Comment by cyqsimon** [comment]
Okay yeah. It was completely my fault. Sorry.

1. I initially thought it wasn't working because I overlooked the `check-inconsistent-struct-field-initializers` option. There was no emission on my actual code because I had field initializers. I tried various versions of rust locally on this code which, of course, yielded the same results.
2. Then I mistakenly thought the example given in docs wasn't working, because I didn't know how to use playground. In my mental model, the items in the "tools" menu merely enable said item, whereas the top-left "run" button executes all the enabled tools. When I started with an example with initializers and clicked "tools > clippy", I got no emissions in the clippy tab due to 1. Then I pasted in the example from the docs and clicked "run", and again saw no emissions so I thought the example must be broken. In reality due to my misunderstanding of the UI, clippy was never rerun at all.

Definitely a big facepalm moment for me. Sorry again for the bother.
